### DISCUSSION TITLE
Network search issues

### DISCUSSION THREAD

**Discussion by xie-yao-oos** [open-webui/open-webui#7540] [Issues]
我在设置中使用了 Searxng url  但是实际使用中发现 他并不能调用到结果原因是: 
﻿
searx.bndkt.io 是该项目最新构建，官方公开的都已经是最新构建了。
﻿
URI： https://searx.bndkt.io/search?q=<query>
﻿
报错: 403 Client Error: FORBIDDEN for url: https://searx.bndkt.io/search?q=12%E6%9C%882%E6%97%A5%E5%8F%91%E7%94%9F%E7%9A%84%E9%87%8D%E8%A6%81%E5%8E%86%E5%8F%B2%E4%BA%8B%E4%BB%B6&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0
﻿
﻿
![Uploading 1733123751254.png…]()

但是我们可以去除一些get 参数 比如 :  &format=json  就可以访问并获得数据:

**Comment by xie-yao-oos** [comment]
requests.exceptions.HTTPError: 403 Client Error: FORBIDDEN for url: https://searx.bndkt.io/search?q=12%E6%9C%882%E6%97%A5%E5%8F%91%E7%94%9F%E7%9A%84%E9%87%8D%E8%A6%81%E5%8E%86%E5%8F%B2%E4%BA%8B%E4%BB%B6&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0
403 Client Error: FORBIDDEN for url: https://searx.bndkt.io/search?q=12%E6%9C%882%E6%97%A5%E5%8F%91%E7%94%9F%E7%9A%84%E9%87%8D%E8%A6%81%E5%8E%86%E5%8F%B2%E4%BA%8B%E4%BB%B6&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0
INFO:     10.*.10.*:56231 - "POST /retrieval/api/v1/process/web/search HTTP/1.1" 400 Bad Request
INFO  [open_webui.apps.openai.main] get_all_models()
ERROR [open_webui.apps.openai.main] Connection error:
INFO  [open_webui.apps.ollama.main] get_all_models()

**Comment by spammenotinoz** [comment]
I found setting a user_agent, improved the reliability of network functions like search and youtube.
eg: "USER_AGENT=Mozilla/5.0 (iPhone; CPU iPhone OS 16_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Mobile/15E148 Safari/604.1"
Perhaps a Linux user_agent would be better, but the above is working for me.

But also a lot of search providers are rate limited.
